server.go

```diff
@@ -436,6 +436,9 @@ func (s *Server) runHandler(
 		ctx = ContextWithShutdownChan(ctx, s.stopChan)
 		ctx = ContextWithQueueName(ctx, queueName)
 
+		acknowledger := NewSafeAcknowledgeHandler(delivery.Acknowledger)
+		delivery.Acknowledger = acknowledger
```

Comment:
Oh nice! TIL the default [Acknowledger](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Acknowledger) on [Delivery](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Delivery)  is just a [Channel](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Channel.Close) which implements `Close` - this looks very smart! :100:

Almost so smart that it would be worth documenting it either here or close to `SafeAcknowledgeHandler`?

```suggestion
		// The Acknowledger on the delivery is just an *amqp.Channel so by using
		// that one we ensure that it can close the channel to restore the connection
		// without closing the queue or discarding messages.
		acknowledger := NewSafeAcknowledgeHandler(delivery.Acknowledger)
		delivery.Acknowledger = acknowledger
```

Reply:
Great idea.